File: platform/extension-sdk/java/src/main/java/io/aksplatform/sdk/SqlDatabase.java
package io.aksplatform.sdk;

public final class SqlDatabase {
    private SqlDatabase() {}

    public static String entraConnectionString(String server, String database) {
        if (server == null || server.isBlank()) {
            throw new IllegalArgumentException("Azure SQL server host is required.");
        }
        if (database == null || database.isBlank()) {
            throw new IllegalArgumentException("Azure SQL database name is required.");
        }

        return String.join(";",
            "jdbc:sqlserver://" + server.trim() + ":1433",
            "database=" + database.trim(),
            "authentication=ActiveDirectoryDefault",
            "encrypt=true",
            "trustServerCertificate=false",
            "loginTimeout=30"
        );
    }
}

File: platform/extension-sdk/dotnet/src/AksPlatform.Sdk/SqlDatabase.cs
using Microsoft.Data.SqlClient;

namespace AksPlatform.Sdk;

public static class SqlDatabase
{
    public static string CreateEntraConnectionString(string server, string database)
    {
        if (string.IsNullOrWhiteSpace(server))
        {
            throw new ArgumentException("Azure SQL server host is required.", nameof(server));
        }

        if (string.IsNullOrWhiteSpace(database))
        {
            throw new ArgumentException("Azure SQL database name is required.", nameof(database));
        }

        var builder = new SqlConnectionStringBuilder
        {
            DataSource = $"tcp:{server.Trim()},1433",
            InitialCatalog = database.Trim(),
            Encrypt = true,
            TrustServerCertificate = false,
            ConnectTimeout = 30,
            Authentication = SqlAuthenticationMethod.ActiveDirectoryDefault
        };

        return builder.ConnectionString;
    }
}
